sandbox-backend/src/main/java/com/exadel/sandbox/team5/dao/DiscountDAO.java

```diff
@@ -34,4 +35,13 @@
             "GROUP BY d.id " +
             "HAVING rate>=(:rate);", nativeQuery = true)
     List<Discount> getByCriteria(@Param("name") String searchText, @Param("rate") int rate);
+
+    @Query(value = """
+            SELECT new com.exadel.sandbox.team5.util.PairSL(d.name, COUNT(o.id))
+            FROM Discount d
+                LEFT JOIN Order o ON d.id=o.discount.id
+            WHERE d.id=o.discount.id
+                GROUP BY d.name
```

Comment:
It is better to use indexed columns for `GROUP BY` block for better performance.

Reply:
done